de-lib/src/main/java/org/iplantc/de/apps/client/presenter/hierarchies/OntologyHierarchiesPresenterImpl.java

```diff
@@ -105,6 +110,42 @@ public void onSuccess(final App appDetails) {
         }
     }
 
+    public class FilteredHierarchyCallback extends PromiseRPC<OntologyHierarchy> {
```

Comment:
`PromiseRPC` is a helper class from GwtQuery to basically mimic the same functionality as AsyncCallback but with Promises.
super gets called at the end of these methods to mark those Promises as complete in the appropriate fashion.  If you call super first, then the `done` method gets called before these methods which would put things out of order in this case.